Java-generator fails on empty string

### Describe the bug

I have a CRD which I've obtained from my k8s cluster by running 

`k get crd inferenceservices.serving.kserve.io -o yaml > file.yaml`

The CRD is quite massive so I'll skip including it here. The java generator fails on parsing an enum that's inside the file:

```
    activeModelState:
      default: Pending
      enum:
      - ""
      - Pending
      - Standby
      - Loading
      - Loaded
      - FailedToLoad
      type: string
```

```
Exception in thread "main" java.lang.AssertionError: A string was unexpectedly empty.
	at com.github.javaparser.utils.Utils.assertNonEmpty(Utils.java:69)
	at com.github.javaparser.ast.nodeTypes.NodeWithSimpleName.setName(NodeWithSimpleName.java:42)
	at io.fabric8.java.generator.nodes.JEnum.generateJava(JEnum.java:95)
	at io.fabric8.java.generator.nodes.JObject.generateJava(JObject.java:168)
	at io.fabric8.java.generator.nodes.JObject.generateJava(JObject.java:168)
	at io.fabric8.java.generator.nodes.JObject.generateJava(JObject.java:168)
```

I've forked the client and can patch this, but I want to confirm that it is desired behavior. On one hand, it does make sense to fail on an empty string, but it's a yaml that's generated by k8s so it should probably just skip that value and not throw an error on it.

Please let me know if you'd like to support this behavior. 



### Fabric8 Kubernetes Client version

SNAPSHOT

### Steps to reproduce

No need

### Expected behavior

The CRD is parsed successfully

### Runtime

minikube

### Kubernetes API Server version

1.24

### Environment

macOS

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

Minikube 1.24.9

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -30,6 +30,7 @@
 * Fix #4574: fromServer has been deprecated - it no longer needs to be called.  All get() operations will fetch the resource(s) from the api server.  If you need the context item that was passed in from a resource, load, or resourceList methods, use the item or items method.
 * Fix #4633: client.run().withRunConfig was deprecated.  Use withNewRunConfig instead.
 * Fix #4663: Config.maxConcurrentRequests and Config.maxConcurrentRequestsPerHost will no longer be used.  Instead they will default to unlimited for all clients.  Due to the ability of the fabric8 client to start long running requests (either websocket or regular http) and how this is treated by the underlying clients you can easily exhaust these values and enter a state where the client is unresponsive without any additional information on what is occurring.
+* Fix #4769: [java-generator] Fix encoding of empty strings as valid enums
 
 ### 6.3.1 (2022-12-15)
 

--- a/java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JEnum.java
+++ b/java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JEnum.java
@@ -17,9 +17,11 @@ package io.fabric8.java.generator.nodes;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
 import com.github.javaparser.ast.body.*;
 import com.github.javaparser.ast.expr.*;
 import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.ast.stmt.ReturnStmt;
 import io.fabric8.java.generator.Config;
 
 import java.util.ArrayList;
@@ -56,7 +58,11 @@ public class JEnum extends AbstractJSONSchema2Pojo {
     if (config.isUppercaseEnums()) {
       ret = ret.toUpperCase(Locale.ROOT);
     }
-    return ret.replaceAll("[\\s/]", "_");
+    if (ret.isEmpty()) {
+      return "_EMPTY";
+    } else {
+      return ret.replaceAll("[\\s/]", "_");
+    }
   }
 
   @Override
@@ -76,6 +82,13 @@ public class JEnum extends AbstractJSONSchema2Pojo {
                     new NameExpr("this." + VALUE),
                     new NameExpr(VALUE),
                     AssignExpr.Operator.ASSIGN)));
+
+    MethodDeclaration getValue = en
+        .addMethod("getValue", Modifier.Keyword.PUBLIC);
+    getValue.setType(JAVA_LANG_STRING);
+    getValue
+        .setBody(new BlockStmt().addStatement(new ReturnStmt(VALUE)));
+    getValue.addAnnotation("com.fasterxml.jackson.annotation.JsonValue");
 
     for (String k : this.values) {
       String constantName;

--- a/java-generator/core/src/test/resources/io/fabric8/java/generator/approvals/ApprovalTest.generate_withValidCrd_shouldGeneratePojos.testJokeCrd.approved.txt
+++ b/java-generator/core/src/test/resources/io/fabric8/java/generator/approvals/ApprovalTest.generate_withValidCrd_shouldGeneratePojos.testJokeCrd.approved.txt
@@ -38,6 +38,11 @@ public class JokeRequestSpec implements io.fabric8.kubernetes.api.model.Kubernet
         Category(java.lang.String value) {
             this.value = value;
         }
+
+        @com.fasterxml.jackson.annotation.JsonValue()
+        public java.lang.String getValue() {
+            return value;
+        }
     }
 
     @com.fasterxml.jackson.annotation.JsonProperty("category")
@@ -71,6 +76,11 @@ public class JokeRequestSpec implements io.fabric8.kubernetes.api.model.Kubernet
 
         Excluded(java.lang.String value) {
             this.value = value;
+        }
+
+        @com.fasterxml.jackson.annotation.JsonValue()
+        public java.lang.String getValue() {
+            return value;
         }
     }
 
@@ -144,6 +154,11 @@ public class JokeRequestStatus implements io.fabric8.kubernetes.api.model.Kubern
 
         State(java.lang.String value) {
             this.value = value;
+        }
+
+        @com.fasterxml.jackson.annotation.JsonValue()
+        public java.lang.String getValue() {
+            return value;
         }
     }
 

--- a/java-generator/it/src/it/enum-ser-deser/src/test/java/io/fabric8/it/certmanager/TestEnumSerialization.java
+++ b/java-generator/it/src/it/enum-ser-deser/src/test/java/io/fabric8/it/certmanager/TestEnumSerialization.java
@@ -41,11 +41,12 @@ class TestEnumSerialization {
     List<CertificateRequestSpec.Usages> usagesList = sample.getSpec().getUsages();
 
     // Assert
-    assertEquals(4, usagesList.size());
-    assertEquals(CertificateRequestSpec.Usages.SIGNING, usagesList.get(0));
-    assertEquals(CertificateRequestSpec.Usages.DIGITAL_SIGNATURE, usagesList.get(1));
-    assertEquals(CertificateRequestSpec.Usages.SERVER_AUTH, usagesList.get(2));
-    assertEquals(CertificateRequestSpec.Usages.S_MIME, usagesList.get(3));
+    assertEquals(5, usagesList.size());
+    assertEquals(CertificateRequestSpec.Usages._EMPTY, usagesList.get(0));
+    assertEquals(CertificateRequestSpec.Usages.SIGNING, usagesList.get(1));
+    assertEquals(CertificateRequestSpec.Usages.DIGITAL_SIGNATURE, usagesList.get(2));
+    assertEquals(CertificateRequestSpec.Usages.SERVER_AUTH, usagesList.get(3));
+    assertEquals(CertificateRequestSpec.Usages.S_MIME, usagesList.get(4));
   }
 
   @Test

--- a/java-generator/it/src/it/enum-ser-deser/src/test/resources/cert-manager.crds.1.7.1.yaml
+++ b/java-generator/it/src/it/enum-ser-deser/src/test/resources/cert-manager.crds.1.7.1.yaml
@@ -140,6 +140,7 @@ spec:
                     description: 'KeyUsage specifies valid usage contexts for keys. See: https://tools.ietf.org/html/rfc5280#section-4.2.1.3      https://tools.ietf.org/html/rfc5280#section-4.2.1.12 Valid KeyUsage values are as follows: "signing", "digital signature", "content commitment", "key encipherment", "key agreement", "data encipherment", "cert sign", "crl sign", "encipher only", "decipher only", "any", "server auth", "client auth", "code signing", "email protection", "s/mime", "ipsec end system", "ipsec tunnel", "ipsec user", "timestamping", "ocsp signing", "microsoft sgc", "netscape sgc"'
                     type: string
                     enum:
+                      - ""
                       - signing
                       - digital signature
                       - content commitment

--- a/java-generator/it/src/it/enum-ser-deser/src/test/resources/sample1.yaml
+++ b/java-generator/it/src/it/enum-ser-deser/src/test/resources/sample1.yaml
@@ -19,9 +19,10 @@ kind: CertificateRequest
 metadata:
   name: my-ca-cr
 spec:
-  request: LS0tL
+  request: dGVzdAo=
   isCA: false
   usages:
+    - ""
     - signing
     - digital signature
     - server auth

